app/sign.js

```diff
@@ -84,6 +90,8 @@ const handleSignUtxo = function(recoveryRequest, key, skipConfirm) {
     throw new Error(`Unsupported coin: ${recoveryRequest.coin}`);
   }
 
+  const isBCH = recoveryRequest.coin === 'bch' || recoveryRequest.coin === 'tbch';
+  const isZEC = recoveryRequest.coin === 'zec' || recoveryRequest.coin === 'tzec';
```

Comment:
Perhaps out of the scope of this change request, but I could see this coin specific logic like this turning into a mess of spaghetti conditionals over time.

An alternative approach we might think about is to take a class based approach ... something along the lines of:

_(excuse the sloppy typescript -- i'm not a js dev) ..._

```typescript
class AbstractCoinSigner {
  handleSignUtxo(recoveryRequest, key, skipConfirm) {
    // keep all existing logic in this fcn

    // provide hooks to allow subclasses to execute custom logic
    transaction = this.optionalInputManipulation(transaction);
  },

   // this fcn is a no-op on the parent class
   optionalInputManipulation(transaction) {
     return transaction;
   },
}

class ZECSigner extends AbstractCoinSigner {

   // The fcn is overidden to apply custom logic for ZEC
   optionalInputManipulation(transaction) { 

     // Add the input values to each input, because input values must be signed
     transaction.ins.forEach(function (input, i) {
       transaction.ins[i].value = recoveryRequest.inputs[i].amount;
     })

     return transaction;
   },
}

// Similar logic for BCH
```

We'd have to do some mapping between the coin assoc'd the wallet being recovered and the classes that should be used in order to sign the recovery tx .... but it keeps the code isolated to its logical components and hopefully easier to reason about over time.

@dacruz21 -- you have thoughts on this?

Reply:
i agree that would be a much cleaner approach

Reply:
My recommendation would be:

* Either close this PR or merge these changes in; let's close the loop on this functionality.
* Create a new PR that starts to break this code up into classes -- @dannybitgo, I'm happy to help you work through this IRL (not much time pressure here -- we just need to get it right. How are we with regards to test coverage?)

Reply:
Pushed some new updates. Turns out not all of that coin-specific stuff was needed. Added test coverage as well. I think this new state is a good state to merge